test/ecto/query/builder/select_test.exs

```diff
@@ -120,6 +120,13 @@ defmodule Ecto.Query.Builder.SelectTest do
       assert [{:{}, [], [:id, escaped_alias]}] == query.select.expr
     end
 
+    test "supports aliasing a selected value in select_merge with selected_as/2" do
```

Comment:
I realized I didn't test select_merge before and added it in instead of opening another PR